internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -561,7 +573,48 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	// We have a watch on both the claim and its composite, so there's no
 	// need to requeue here.
 	cm.SetConditions(xpv1.Available())
-	return reconcile.Result{Requeue: false}, errors.Wrap(r.client.Status().Update(ctx, cm), errUpdateClaimStatus)
+	return reconcile.Result{Requeue: false}, errors.Wrap(r.client.Status().Update(ctx, cm, r.fieldOwner), errUpdateClaimStatus)
+}
+
+// For the background, check https://github.com/kubernetes/kubernetes/issues/99003
+// Managed fields owner key is a pair (the manager name, the used operation).
+// Initially, we create a composite using CREATE request, because that is the only
+// way to figure out if the generated name is already taken.
+// If the operation is successful, managed fields operation will be set to "Update".
+// Later patches are going to produce another managed fields set for the same manager,
+// but with a different operation "Apply".
+// Such configuration would prevent removal of claim fields on the composite, i.e.
+// it would only remove the ownership.
+// In order to fix that, we need to manually change operation to "Apply",
+// before the first composite patch is sent to k8s api server.
+// Returns true if the ownership was fixed.
+func (r *Reconciler) maybeFixFieldOwnership(obj *unstructured2.Unstructured, fixSubresource bool) bool {
+	mfs := obj.GetManagedFields()
+	fixed := false
+	for i := range mfs {
+		if mfs[i].Subresource != "" && !fixSubresource {
+			continue
+		}
+		if mfs[i].Manager == string(r.fieldOwner) && mfs[i].Operation == "Update" {
+			mfs[i].Operation = "Apply"
+			obj.SetManagedFields(mfs)
+			fixed = true
+			continue
+		}
+
+		// Even if the server-side apply is not used, api server derives manager name
+		// from the submitted user agent (see net/http/request.go).
+		// After crossplane update, we need to replace the ownership so that
+		// field removals can be propagated properly.
+		if mfs[i].Manager == "Go-http-client" && mfs[i].Operation == "Update" {
```

Comment:
Are we sure that this will migrate old versions? 

Just deployed a claim/composite with `v1.14.2` and below is what I am seeing:

```
      manager: crossplane
      operation: Update
```

<details>

<summary>composite</summary>

```yaml
managedFields:
    - apiVersion: nop.example.org/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:finalizers:
            .: {}
            v:"composite.apiextensions.crossplane.io": {}
          f:generateName: {}
          f:labels:
            .: {}
            f:crossplane.io/claim-name: {}
            f:crossplane.io/claim-namespace: {}
            f:crossplane.io/composite: {}
        f:spec:
          .: {}
          f:claimRef:
            .: {}
            f:apiVersion: {}
            f:kind: {}
            f:name: {}
            f:namespace: {}
          f:compositionRef:
            .: {}
            f:name: {}
          f:compositionRevisionRef:
            .: {}
            f:name: {}
          f:compositionUpdatePolicy: {}
          f:coolField: {}
          f:resourceRefs: {}
      manager: crossplane
      operation: Update
      time: "2023-12-05T06:54:58Z"
    - apiVersion: nop.example.org/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          .: {}
          f:conditions:
            .: {}
            k:{"type":"Ready"}:
              .: {}
              f:lastTransitionTime: {}
              f:reason: {}
              f:status: {}
              f:type: {}
            k:{"type":"Synced"}:
              .: {}
              f:lastTransitionTime: {}
              f:reason: {}
              f:status: {}
              f:type: {}
          f:nginx:
            .: {}
            f:conditions: {}
            f:observedGeneration: {}
            f:replicas: {}
            f:unavailableReplicas: {}
            f:updatedReplicas: {}
      manager: crossplane
      operation: Update
      subresource: status
      time: "2023-12-05T06:54:59Z"
```

</details>

<details>

<summary>claim</summary>

```yaml
managedFields:
    - apiVersion: nop.example.org/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:finalizers:
            .: {}
            v:"finalizer.apiextensions.crossplane.io": {}
        f:spec:
          f:compositionRef:
            .: {}
            f:name: {}
          f:compositionRevisionRef:
            .: {}
            f:name: {}
          f:compositionUpdatePolicy: {}
          f:resourceRef:
            .: {}
            f:apiVersion: {}
            f:kind: {}
            f:name: {}
      manager: crossplane
      operation: Update
      time: "2023-12-05T06:54:58Z"
    - apiVersion: nop.example.org/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            .: {}
            f:kubectl.kubernetes.io/last-applied-configuration: {}
        f:spec:
          .: {}
          f:compositeDeletePolicy: {}
          f:coolField: {}
      manager: kubectl-client-side-apply
      operation: Update
      time: "2023-12-05T06:54:58Z"
    - apiVersion: nop.example.org/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          .: {}
          f:conditions:
            .: {}
            k:{"type":"Ready"}:
              .: {}
              f:lastTransitionTime: {}
              f:reason: {}
              f:status: {}
              f:type: {}
            k:{"type":"Synced"}:
              .: {}
              f:lastTransitionTime: {}
              f:reason: {}
              f:status: {}
              f:type: {}
          f:nginx:
            .: {}
            f:availableReplicas: {}
            f:conditions: {}
            f:observedGeneration: {}
            f:readyReplicas: {}
            f:replicas: {}
            f:unavailableReplicas: {}
            f:updatedReplicas: {}
      manager: crossplane
      operation: Update
      subresource: status
      time: "2023-12-05T06:55:59Z"
```

</details>

Reply:
Good catch! I have fixed the issue and updated the e2e tests for check explicitly for that.